### PR TITLE
Remove dependencies on enum_primitive and num

### PR DESCRIPTION
**Pull request by glowcoil**
Depending on these packages brings in 9 dependencies in total, including num-complex and num-rational, and adds ~50% to compile times for me (33s up to 49s for release builds). This doesn't seem worth it to me (although I'm willing to change my mind, or write a replacement, if others disagree).

### CODE REVIEW

**Comment by bvssvni** [comment]
Notice that casting with `as` can be done without using num traits, however it would be nice to have tests to make sure it does not break.

File: src/lib.rs [hunk 1]
```diff
@@ -327,29 +322,6 @@ pub enum MouseButton {
     Button8                = ffi::MOUSE_BUTTON_8,
 }
 
-// We can't use `enum_from_primitive!` for MouseButton due to
-// https://github.com/andersk/enum_primitive-rs/issues/2
-// We implement FromPrimitive manually instead.
-impl num::FromPrimitive for MouseButton {
```
Comment (aloucks) [inline src/lib.rs]: Why not keep the `from_i64` and `from_u64` as non-trait methods?

**Comment by bvssvni** [comment]
If the issue with keeping `from_i64` and `from_u64` can be resolved, then I think we can merge this.

File: src/lib.rs (hunk 1 above)
Comment (glowcoil) [reply src/lib.rs]: After thinking about this, I decided I would prefer to replace them (and the other FromPrimitive implementations) with a `from_i32` that does a range check and a transmute.

**Comment by bvssvni** [comment]
Looks good. Merging.

**Comment by bvssvni** [comment]
Thanks!

**Pull request merged**